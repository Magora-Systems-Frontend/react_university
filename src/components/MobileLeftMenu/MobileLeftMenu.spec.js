import React from 'react';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import { MobileLeftMenu as Component } from './index';

configure({ adapter: new Adapter() });

const initialState = { languageState: { language: 'EN' } };

describe('component snapshot', function() {
  it('capturing snapshot of Footer', () => {
    const renderedValue = renderer.create(<Component languageState={initialState.languageState} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('test dump component', function() {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Component languageState={initialState.languageState} />);
  });

  it('render the dumb component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('check change props', () => {
    wrapper.setProps({ show: true });
    expect(wrapper.find('.mobile-menu__overlay_show')).toHaveLength(1);
  });
});
