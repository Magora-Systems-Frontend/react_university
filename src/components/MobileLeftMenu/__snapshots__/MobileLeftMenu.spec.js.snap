// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component snapshot capturing snapshot of Footer 1`] = `
<div
  className="mobile-menu_wrapper"
>
  <div
    className="mobile-menu__overlay"
    onClick={[Function]}
  />
  <nav
    className="mobile-nav"
  >
    <button
      className="btn-action btn_mobile-login"
      onClick={[Function]}
    >
      Sign up / Log in
    </button>
    <hr
      className="mobile-nav__divider"
    />
    <div
      className="mobile-nav__menu"
    >
      <div
        className="mobile-nav__menu-title"
      >
        MORE PRODUCTS
      </div>
      <ul
        className="mobile-nav__menu-list"
      >
        <li>
          <a
            className="mobile-nav__menu-item"
            href={null}
          >
            Teach
          </a>
        </li>
        <li>
          <a
            className="mobile-nav__menu-item"
            href={null}
          >
            Business
          </a>
        </li>
        <li>
          <a
            className="mobile-nav__menu-item"
            href={null}
          >
            Get the app
          </a>
        </li>
        <li>
          <a
            className="mobile-nav__menu-item"
            href={null}
          >
            Help
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
